### PR TITLE
drivers: modem: Add support for different line endings

### PR DESCRIPTION
**Pull request by tsvehagen**

### CODE REVIEW

File: drivers/modem/modem_cmd_handler.h [hunk 1]
```diff
@@ -75,6 +75,8 @@ struct modem_cmd_handler_data {
 
 	int last_error;
 
+	char *eol;
```
Comment (jukkar) [inline drivers/modem/modem_cmd_handler.h]: This seems to be always a const string, so perhaps adding `const char *eol`? here.

File: drivers/modem/modem_cmd_handler.c [hunk 2]
```diff
@@ -388,7 +388,10 @@ static int _modem_cmd_send(struct modem_iface *iface,
 	}
 
 	iface->write(iface, buf, strlen(buf));
-	iface->write(iface, "\r", 1);
+
+	if (data->eol != NULL) {
+		iface->write(iface, data->eol, strlen(data->eol));
```
Comment (jukkar) [inline drivers/modem/modem_cmd_handler.c]: What about calculating the length only once and storing the length in `modem_cmd_handler_data` struct. The send function is called all the time and it feels useless to calculate the length every time even if it is not changing.

**Review by jukkar — CHANGES_REQUESTED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by mike-scott** [comment]
LGTM.  Let's get this in and I'll rebase https://github.com/zephyrproject-rtos/zephyr/pull/22149 on top.

**Review by mike-scott — APPROVED** [review]

**Pull request merged**